R/mcmc.R

```diff
@@ -1,17 +1,52 @@
 
-#' Title - TODO
+#' Fit the base imputation model using a Bayesian approach
+#'
+#' @description
+#' `fit_mcmc()` fits the base imputation model using a Bayesian approach. This is done through a MCMC method that is implemented in `stan`
+#' and is run by using the function [rstan::sampling()].
+#' The function returns the draws from the posterior distribution of the model parameters
+#' and the `stanfit` object. Additionally it performs multiple diagnostics checks of the chain
+#' and returns warnings in case of any detected issues.
+#'
+#' @param designmat The design matrix of the fixed effects.
+#' @param outcome The response variable. Must be numeric.
+#' @param group The group variable.
+#' @param subjid Character vector containing the subjects IDs.
+#' @param visit The visit variable.
+#' @param same_cov Logical. If `TRUE` the model assumes the same covariance matrix for each group.
+#' If `FALSE` a different covariance matrix for each group is assumed.
+#' @param n_imputations Integer number corresponding to the draws from the posterior distribution needed.
+#' It corresponds to the number of imputations to be done in the multiple imputation procedure.
+#' @param burn_in Integer number corresponding to the number of initial MCMC iterations to be discarded. See details.
+#' @param burn_between Integer number corresponding to the thinning of the chain. It is the number
+#' of iterations that are discarded between two consecutive draws. See details.
+#' @param verbose Logical. If `TRUE` the chain flow of information is displayed in the console.
+#' @param seed Integer number to set the seed for reproducibility of the MCMC draws.
+#'
+#' @details
+#' The Bayesian model assumes a multivariate normal likelihood function and weakly-informative
+#' priors for the model parameters: in particular, uniform priors are assumed for the regression
+#' coefficients and inverse-Wishart priors for the covariance matrices.
+#' The chain is initialized using the REML parameter estimates from MMRM as starting values.
+#'
+#' The function performs the following steps:
+#' 1. Fit MMRM using a REML approach.
+#' 2. Prepare the input data for the MCMC fit as described in the `data{}` block of the Stan file. See [prepare_stan_data()] for details.
+#' 3. Run the MCMC according the input arguments and using as starting values the REML parameter estimates
+#' estimated at point 1.
+#' 4. Performs diagnostics checks of the MCMC. See [check_mcmc()] for details.
+#' 5. Extract the draws from the model fit.
+#'
+#' The chains perform `n_imputations` draws by keeping one every `burn_between` iterations. Additionally
+#' the first `burn_in` iterations are discarded. The total number of iterations will then be `burn_in + burn_between*n_imputations`.
+#' The `burn_in` aims to perform draws from the stationary distribution of the Markov Chain.
```

Comment:
Maybe reword this slightly?

"the purpose of `burn_in` is to ensure that the samples that are drawn are drawn from the stationary ..." ?

maybe unnecessary though